rope/base/utils.py

```diff
@@ -81,3 +81,48 @@ def __call__(self, *args, **kwds):
         if len(self.cache) > self.count:
             del self.cache[0]
         return result
+
+class memoize(object):
```

Comment:
There is [new version of class _Cache()](https://github.com/python-rope/rope/commit/fc7412837a81201de9711a585a704c0b0fce6231#diff-f52bb21de086b8892647fec5277b4b23R94) with LRU but without list. Much faster gets value.